### PR TITLE
Add an installer script

### PR DESCRIPTION
**Pull request by Hassunaama**
This includes Bepinex, 6Dash and 3DashTools install.

### CODE REVIEW

**Comment by cgytrus** [comment]
why does it need admin perms?

**Comment by Hassunaama** [comment]
> why does it need admin perms?

It cannot run the Bepinex script without them because of windows policy's

File: 3DashTools-Install.bat [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+:bepinex_install
+call powershell -command Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope LocalMachine >nul
+call powershell -command (new-object System.Net.WebClient).DownloadFile('https://gist.githubusercontent.com/cgytrus/29085a6bf179893666316a36e1c92bf6/raw/bepinex-installer.ps1', 'bepinex-installer.ps1') >nul
+powershell .\bepinex-installer.ps1 >nul
+del /f /q bepinex-installer.ps1
+set /a progress=3
+goto install_5
+
+:6dash_install
+call powershell -command (new-object System.Net.WebClient).DownloadFile('https://github.com/cgytrus/SixDash/releases/download/v0.3.0/SixDash-0.3.0.zip', 'sixdash.zip') >nul
```
Comment (cgytrus) [inline 3DashTools-Install.bat:25]: don't hardcode the versions, use the github api to get the latest version like in the bepinex installer script

Comment (cgytrus) [reply 3DashTools-Install.bat:25]: same on line 32

File: 3DashTools-Install.bat [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    goto begin_main
+) else (
+    echo Error: Administrator permissions needed.
+    pause
+    exit
+)
+:begin_main
+cls
+set current_time=%time:~0,5%
+if /i "%current_time%" GEQ " 5:00" if /i "%current_time%" LSS "13:00" echo Good morning %username%! Welcome to the 3DashTools Installer.
```
Comment (cgytrus) [inline 3DashTools-Install.bat]: why is this even a thing in this script

File: 3DashTools-Install.bat [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+goto install_5
+
+:install_5
+cls
+set current_time=%time:~0,5%
+if /i "%current_time%" GEQ " 5:00" if /i "%current_time%" LSS "13:00" echo Good morning %username%! Welcome to the 3DashTools Installer.
+if /i "%current_time%" GEQ "13:00" if /i "%current_time%" LSS "18:00" echo Good afternoon %username%! Welcome to the 3DashTools Installer.
+if /i "%current_time%" GEQ "18:00" if /i "%current_time%" LEQ "23:59" echo Good evening %username%! Welcome to the 3DashTools Installer.
+if /i "%current_time%" GEQ " 0:00" if /i "%current_time%" LSS " 5:00" echo Good evening %username%! Welcome to the 3DashTools Installer.
+echo.
```
Comment (cgytrus) [inline 3DashTools-Install.bat:95]: copy-pasted code

File: 3DashTools-Install.bat [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+if "%progress%"=="8" goto final
+if "%progress%"=="9" goto end
+
+goto install_5
+:end
+cls
+echo Thanks for using 3DashTools Installer!
+echo Have fun with 3DashTools!
+echo.
+echo Feel free to close this window.
```
Comment (cgytrus) [inline 3DashTools-Install.bat]: the texts in this are generally weird idk

File: 3DashTools-Install.bat [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+echo.
+echo Press any key to open 3Dash.
+pause >nul
+3Dash.exe
```
Comment (cgytrus) [inline 3DashTools-Install.bat]: wouldnt do that

**Review by cgytrus — CHANGES_REQUESTED** [review]

**Comment by cgytrus** [comment]
generally i dont rly think it's a good idea to have an installer script, it's better to put effort into adding 3dash support to [r2mm](https://github.com/ebkr/r2modmanPlus)

**Comment by cgytrus** [comment]
the problem with that, however, is that it won't work unless the game is on a supported store, itch.io is not one of them
we'll probably have to work with ebkr to either add support for itch.io (which would actually require players to use the itch app to get the game), or add proper support for games from unsupported stores

File: 3DashTools-Install.bat (hunk 2 above)
Comment (Hassunaama) [reply 3DashTools-Install.bat]: Fixed?

File: 3DashTools-Install.bat (hunk 3 above)
Comment (Hassunaama) [reply 3DashTools-Install.bat:95]: Fixed.

File: 3DashTools-Install.bat (hunk 4 above)
Comment (Hassunaama) [reply 3DashTools-Install.bat]: Fixed.

File: 3DashTools-Install.bat (hunk 5 above)
Comment (Hassunaama) [reply 3DashTools-Install.bat]: Fixed.

**Comment by cgytrus** [comment]
noticed you made more changes
my comments still apply, i don't really want an installer script
the reason i'm not closing this pr is because i thought you could add more to the conversation

**Comment by Hassunaama** [comment]
> generally i dont rly think it's a good idea to have an installer script, it's better to put effort into adding 3dash support to [r2mm](https://github.com/ebkr/r2modmanPlus)

As you want.

**Pull request closed without merging**